server/store/database/access_token.go

```diff
@@ -0,0 +1,80 @@
+package database
+
+import (
+	"context"
+	"time"
+
+	"github.com/charmbracelet/soft-serve/server/db"
+	"github.com/charmbracelet/soft-serve/server/db/models"
+	"github.com/charmbracelet/soft-serve/server/store"
+)
+
+type accessTokenStore struct{}
+
+var _ store.AccessTokenStore = (*accessTokenStore)(nil)
+
+// CreateAccessToken implements store.AccessTokenStore.
+func (s *accessTokenStore) CreateAccessToken(ctx context.Context, h db.Handler, name string, userID int64, token string, expiresAt time.Time) (models.AccessToken, error) {
+	query := `INSERT INTO access_tokens (name, user_id, token,`
+	vquery := `VALUES (?, ?, ?,`
+	values := []interface{}{name, userID, token}
+	if !expiresAt.IsZero() {
+		query += ` expires_at,`
+		vquery += ` ?,`
+		values = append(values, expiresAt)
+	}
+
+	query += ` created_at, updated_at)`
+	vquery += ` CURRENT_TIMESTAMP, CURRENT_TIMESTAMP)`
+	query = h.Rebind(query + " " + vquery)
```

Comment:
💅🏻 this would probably be more easy to understand if split into two queries, even if it would add a little repetition